xmtp_mls/src/groups/message_history.rs

```diff
@@ -65,6 +76,22 @@ pub enum MessageHistoryError {
     Client(#[from] ClientError),
     #[error("group error: {0}")]
     Group(#[from] GroupError),
+    #[error("request ID of reply does not match request")]
+    ReplyRequestIdMismatch,
+    #[error("reply already processed")]
+    ReplyAlreadyProcessed,
+    #[error("no pending request to reply to")]
+    NoPendingRequest,
+    #[error("no reply to process")]
+    NoReplyToProcess,
+    #[error("generic: {0}")]
+    Generic(String),
+    #[error("missing history sync url")]
+    MissingHistorySyncUrl,
+    #[error("invalid history message payload")]
+    InvalidPayload,
+    #[error("invalid history bundle url")]
+    InvalidBundleUrl,
 }
```

Comment:
Can be done in a follow up, but we're going to need to define which of these error variants are retryable or not. Probably just the `Storage` directly, but `Client` and `Group` should defer to those error type's `RetryableError` implementation to see if the underlying error is retryable too.